tools/mason/MasonExternal.chpl

```diff
@@ -717,3 +730,22 @@ proc uninstallSpkg(args: [?d] string) throws {
     throw new owned MasonError("Package could not be uninstalled");
   }
 }
+
+// compare two spack version strings in format `major.minor.patch`
+// expects matching number of elements, separated by a `.`
+// e.g 0.15.0 or 0.16.2
+// returns -1 if foundVersion < expectedVersion
+// return 0 if versions are the same
+// return 1 if foundVersion > expectedVersion (not likely)
+proc compareSpackVersion(foundVersion:string,
```

Comment:
Is the `VersionInfo` robust enough that we could just make `spackVersion` into a `VersionInfo` instead of a string?

Reply:
I believe so. I'll work on making that change.

Reply:
I converted the spackVersion into an instance of VersionInfo and added a method to VersionInfo to check if another version is compatible. Compatible versions share the same major and minor version but could have a higher bug/patch version.